firebase-app-distribution/src/test/java/com/google/firebase/appdistribution/FirebaseAppDistributionTest.java

```diff
@@ -0,0 +1,154 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.robolectric.Shadows.shadowOf;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Intent;
+import android.content.pm.ResolveInfo;
+import android.net.Uri;
+import android.os.Bundle;
+import androidx.test.core.app.ApplicationProvider;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.installations.FirebaseInstallationsApi;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.shadows.ShadowActivity;
+import org.robolectric.shadows.ShadowAlertDialog;
+import org.robolectric.shadows.ShadowPackageManager;
+
+@RunWith(RobolectricTestRunner.class)
+public class FirebaseAppDistributionTest {
+
+  private FirebaseApp firebaseApp;
+  private FirebaseAppDistribution firebaseAppDistribution;
+
+  @Mock private FirebaseInstallationsApi mockFirebaseInstallations;
+  @Mock private Bundle mockBundle;
+  @Mock SignInResultActivity mockSignInResultActivity;
+
+  public static final String TEST_API_KEY = "AIzaSyabcdefghijklmnopqrstuvwxyz1234567";
+  public static final String TEST_APP_ID_1 = "1:123456789:android:abcdef";
+  public static final String TEST_PROJECT_ID = "777777777777";
+  public static final String TEST_FID_1 = "cccccccccccccccccccccc";
+  public static final String TEST_URL =
+      String.format(
+          "https://appdistribution.firebase.dev/nba/pub/apps/%s/installations/%s/buildalerts?appName=com.google.firebase.appdistribution.test",
+          TEST_APP_ID_1, TEST_FID_1);
+
+  private TestActivity activity;
+  private ShadowActivity shadowActivity;
+  private ShadowPackageManager shadowPackageManager;
+
+  public static class TestActivity extends Activity {}
+
+  @Before
+  public void setup() {
+
+    MockitoAnnotations.initMocks(this);
+
+    FirebaseApp.clearInstancesForTest();
+
+    firebaseApp =
+        FirebaseApp.initializeApp(
+            ApplicationProvider.getApplicationContext(),
+            new FirebaseOptions.Builder()
+                .setApplicationId(TEST_APP_ID_1)
+                .setProjectId(TEST_PROJECT_ID)
+                .setApiKey(TEST_API_KEY)
+                .build());
+
+    firebaseAppDistribution = new FirebaseAppDistribution(firebaseApp, mockFirebaseInstallations);
+
+    when(mockFirebaseInstallations.getId()).thenReturn(Tasks.forResult(TEST_FID_1));
+
+    shadowPackageManager =
+        shadowOf(ApplicationProvider.getApplicationContext().getPackageManager());
+    activity = Robolectric.buildActivity(TestActivity.class).create().get();
+    shadowActivity = shadowOf(activity);
+  }
+
+  @Test
```

Comment:
This is a minor suggestion - A good rule of thumb for test method names is `methodBeingTests_condition_expectedResult` so you could rename this to be 
`signInTester_whenDialogConfirmedAndChromeAvailable_opensCustomTab()` (basically remove the extra _ )

Same comment for method names below